x-pack/plugins/maps/public/classes/sources/es_geo_line_source/es_geo_line_source.tsx

```diff
@@ -0,0 +1,352 @@
+    }
+    return {
+      ...normalizedDescriptor,
+      type: SOURCE_TYPES.ES_GEO_LINE,
+      geoField: normalizedDescriptor.geoField!,
+      splitField: normalizedDescriptor.splitField!,
+      sortField: normalizedDescriptor.sortField!,
+    };
+  }
+
+  readonly _descriptor: ESGeoLineSourceDescriptor;
+
+  constructor(descriptor: Partial<ESGeoLineSourceDescriptor>, inspectorAdapters?: Adapters) {
+    const sourceDescriptor = ESGeoLineSource.createDescriptor(descriptor);
+    super(sourceDescriptor, inspectorAdapters, true);
+    this._descriptor = sourceDescriptor;
+  }
+
+  renderSourceSettingsEditor({ onChange }: SourceEditorArgs) {
+    return (
+      <UpdateSourceEditor
+        indexPatternId={this.getIndexPatternId()}
+        onChange={onChange}
+        metrics={this._descriptor.metrics}
+        splitField={this._descriptor.splitField}
+        sortField={this._descriptor.sortField}
+      />
+    );
+  }
+
+  getSyncMeta() {
+    return {
+      splitField: this._descriptor.splitField,
+      sortField: this._descriptor.sortField,
+    };
+  }
+
+  async getImmutableProperties(): Promise<ImmutableSourceProperty[]> {
+    let indexPatternTitle = this.getIndexPatternId();
+    try {
+      const indexPattern = await this.getIndexPattern();
+      indexPatternTitle = indexPattern.title;
+    } catch (error) {
+      // ignore error, title will just default to id
+    }
+
+    return [
+      {
+        label: getDataSourceLabel(),
+        value: geoLineTitle,
+      },
+      {
+        label: i18n.translate('xpack.maps.source.esGeoLine.indexPatternLabel', {
+          defaultMessage: 'Index pattern',
+        }),
+        value: indexPatternTitle,
+      },
+      {
+        label: i18n.translate('xpack.maps.source.esGeoLine.geospatialFieldLabel', {
+          defaultMessage: 'Geospatial field',
+        }),
+        value: this._descriptor.geoField,
+      },
+    ];
+  }
+
+  _createSplitField(): IField {
+    return new ESDocField({
+      fieldName: this._descriptor.splitField,
+      source: this,
+      origin: FIELD_ORIGIN.SOURCE,
+      canReadFromGeoJson: true,
+    });
+  }
+
+  getFieldNames() {
+    return [
+      ...this.getMetricFields().map((esAggMetricField) => esAggMetricField.getName()),
+      this._descriptor.splitField,
+      this._descriptor.sortField,
+    ];
+  }
+
+  async getFields(): Promise<IField[]> {
+    return [...this.getMetricFields(), this._createSplitField()];
+  }
+
+  getFieldByName(name: string): IField | null {
+    return name === this._descriptor.splitField
+      ? this._createSplitField()
+      : this.getMetricFieldForName(name);
+  }
+
+  isGeoGridPrecisionAware() {
+    return false;
+  }
+
+  showJoinEditor() {
+    return false;
+  }
+
+  async getGeoJsonWithMeta(
+    layerName: string,
+    searchFilters: VectorSourceRequestMeta,
+    registerCancelCallback: (callback: () => void) => void,
+    isRequestStillActive: () => boolean
+  ): Promise<GeoJsonWithMeta> {
```

Comment:
Should we throw here if user does not have a gold+ license?

e.g. after a trial expires, these layers will fail to fetch data.

We could provide a more specific human readable error, and not call ES